smartcontract/smart_contract.go

```diff
@@ -157,7 +158,8 @@ func (this *SmartContract) NewExecuteEngine(code []byte, txtype ctypes.Transacti
 			Height:     this.Config.Height,
 			BlockHash:  this.Config.BlockHash,
 			PreExec:    this.PreExec,
-			GasLimit:   this.Gas,
+			GasLimit:   &this.Gas,
+			GasFactor:  this.GasTable[config.WASM_GAS_FACTOR],
```

Comment:
check `this.GasTable[config.WASM_GAS_FACTOR]` is not 0, and replace it with the default value to avoid divide zero panic.